l1-contracts/contracts/bridgehub/STMDeploymentTracker.sol

```diff
@@ -49,7 +50,10 @@ contract STMDeploymentTracker is ISTMDeploymentTracker, ReentrancyGuard, Ownable
         // solhint-disable-next-line gas-custom-errors
 
         require(BRIDGE_HUB.stateTransitionManagerIsRegistered(_stmAddress), "STMDT: stm not registered");
-        SHARED_BRIDGE.setAssetHandlerAddressInitial(bytes32(uint256(uint160(_stmAddress))), address(BRIDGE_HUB));
+        IAssetRouterBase(address(SHARED_BRIDGE)).setAssetHandlerAddress(
```

Comment:
The Shared_Bridge can be IAssetRouterBase. (Also it should be Asset_Router, but we can also do the renaming later)

Reply:
I fixed this is other places, but for L1AssetRouter I can't inherit IAssetRouterBase, as it results in circular dependency

Reply:
What causes the circularity? It should not be the case

Reply:
Import of the bridgehub, 
I will resolve it tomorrow